### PR TITLE
Project structure

### PR DESCRIPTION
**Pull request by cschreep**
This request introduces a number of changes our project structure to ultimately simplify, harden, and streamline  our installation, build, and deployment processes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ENV UID 1000
+
+RUN adduser --disabled-login --home $HOME --shell /bin/bash --uid $UID $USER
+
+USER $USER
+WORKDIR $HOME
+
+ENV PATH=$HOME/.local/bin:$PATH
+ENV ELASTIC_API_HOST=
+ENV ELASTIC_API_PORT=
```
Comment (andrew-rti) [inline Dockerfile:25]: I know these existed before the PR, but, is there a good reason to have this defined but not set?

File: bin/dug [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#!/bin/bash
```
Comment (andrew-rti) [inline bin/dug:1]: nit: `#!/usr/bin/env bash`

**Review by andrew-rti — APPROVED** [review]

**Review by YaphetKG — APPROVED** [review]
great stuff, was able to run tests and crawl according to the readme.

File: Dockerfile (hunk 1 above)
Comment (cschreep) [reply Dockerfile:25]: I don't actually know what the motivation was. I just tested removing them and everything ran smoothly 🤷

**Pull request merged**